dockerfile_parse/parser.py

```diff
@@ -92,9 +92,12 @@ def __init__(self, path=None, cache_content=False, env_replace=True, parent_env=
 
         self.env_replace = env_replace
 
-        if parent_env:
+        if isinstance(parent_env, dict):
             logger.debug("Setting inherited parent image ENV vars: %s", parent_env)
             self.parent_env = parent_env
+        elif parent_env != None:
+            logger.debug("Invalid parent image ENV provided: %s", parent_env)
```

Comment:
If we want to do that, maybe we should just `assert isinstance(parent_env, dict)`.